### ISSUE TITLE
DBeaver for Greenplum: support of replicated tables if PK

### ISSUE THREAD

**Issue by lintzc** [dbeaver/dbeaver#12414]
#### System information: 
- Operating system (distribution) and version: Mac OS
- DBeaver version: Version 21.0.4.202105021514
- Additional extensions

#### Connection specification:
- Database name and version: Greenplum 6
- Driver name: Greenplum
- Do you use tunnels or proxies (SSH, SOCKS, etc)? no

#### Describe the problem you're observing:
If a table is defined as replicated and has a primary key, DBeaver displays the DDL as distributed by the field of the Primary Key, instead of replicated.

#### Steps to reproduce, if exist:

`create table t_replicated_pk (a int primary key, b text) distributed replicated;
`
With psql:
```
dbtest=# d+ t_replicated_pk
                    Table "public.t_replicated_pk"
 Column |  Type   | Modifiers | Storage  | Stats target | Description
--------+---------+-----------+----------+--------------+-------------
 a      | integer | not null  | plain    |              |
 b      | text    |           | extended |              |
Indexes:
    "t_replicated_pk_pkey" PRIMARY KEY, btree (a)
Distributed Replicated

```
With DBeaver:
```
-- Drop table

-- DROP TABLE public.t_replicated_pk;

CREATE TABLE public.t_replicated_pk (
        a int4 NOT NULL,
        b text NULL,
        CONSTRAINT t_replicated_pk_pkey PRIMARY KEY (a)
)
DISTRIBUTED BY (a);
```

**Comment by LonwoLonwo** [comment]
Thanks for the bug report!

**Comment by LonwoLonwo** [comment]
"When creating a table, there is an additional clause to declare the Greenplum Database distribution policy. If a DISTRIBUTED BY, DISTRIBUTED RANDOMLY, or DISTRIBUTED REPLICATED clause is not supplied, then Greenplum Database assigns a hash distribution policy to the table using either the PRIMARY KEY (if the table has one) or the first column of the table as the distribution key. Columns of geometric or user-defined data types are not eligible as Greenplum distribution key columns. If a table does not have a column of an eligible data type, the rows are distributed based on a round-robin or random distribution. To ensure an even distribution of data in your Greenplum Database system, you want to choose a distribution key that is unique for each record, or if that is not possible, then choose DISTRIBUTED RANDOMLY."

https://gpdb.docs.pivotal.io/6-16/ref_guide/sql_commands/CREATE_TABLE.html

**Comment by lintzc** [comment]
You are using a very old version GPDB 4.3.30.
Now, most of the customers are using GPDB6: https://gpdb.docs.pivotal.io/6-16/ref_guide/sql_commands/CREATE_TABLE.html

You can see that there is another kind of distribution: REPLICATED.

Cyrille



________________________________
De : Anastasiya ***@***.***>
Envoyé : mercredi 12 mai 2021 10:26
À : dbeaver/dbeaver ***@***.***>
Cc : Cyrille Lintz (Pivotal) ***@***.***>; Author ***@***.***>
Objet : Re: [dbeaver/dbeaver] DBeaver for Greenplum: support of replicated tables if PK (#12414)


"When creating a table, there is an additional clause to declare the Greenplum Database distribution policy. If a DISTRIBUTED BY or DISTRIBUTED RANDOMLY clause is not supplied, then Greenplum assigns a hash distribution policy to the table using either the PRIMARY KEY (if the table has one) or the first column of the table as the distribution key. Columns of geometric or user-defined data types are not eligible as Greenplum distribution key columns. If a table does not have a column of an eligible data type, the rows are distributed based on a round-robin or random distribution. To ensure an even distribution of data in your Greenplum Database system, you want to choose a distribution key that is unique for each record, or if that is not possible, then choose DISTRIBUTED RANDOMLY."

https://gpdb.docs.pivotal.io/43300/ref_guide/sql_commands/CREATE_TABLE.html<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgpdb.docs.pivotal.io%2F43300%2Fref_guide%2Fsql_commands%2FCREATE_TABLE.html&data=04%7C01%7Cclintz%40vmware.com%7C953666e11e9c49d0dd9b08d9151f95c6%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C637564047687088025%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=3qwDXFxQTrlaxD5%2FxBrXZRQzvJd%2FdVyx4yWZWDGhI3w%3D&reserved=0>

**Comment by LonwoLonwo** [comment]
Fixed in DBeaver 21.0.5
Will be available on 2021-05-17

**Comment by uslss** [comment]
verified
